### ISSUE TITLE
Conflicting $record when using widget in a resource

### ISSUE THREAD

**Issue by jigar-dhulla** [saade/filament-fullcalendar#209]
I am trying to use this widget in a resource and `$record` property in `InteractsWithRecords` trait is conflicting with the resource record.

Our `fetchEvents()` implementation to fetch the events uses `$this->record` which populates the resource record correctly in the beginning. But when we click on any event, value of $this->record is overwritten by `Event` record.

I see a change was made in #41 , but it seems that was for an older version.

**Comment by PrasadChinwal** [comment]
Facing the same issue.

**Comment by saade** [comment]
Yes, unfortunately this is a know issue. I'm rewriting this implementation for v4

**Comment by PrasadChinwal** [comment]
@saade Do we have a workaround to not throw the error?

**Comment by kamilst96** [comment]
Can't actively vouch for it, but this is my current workaround.

Install composer package:
`composer require cweagans/composer-patches`

Add in composer.json:
```
"extra": {
        "patches": {
            "saade/filament-fullcalendar": {
                "Fix conflicting record": "https://github.com/kamilst96/filament-fullcalendar/commit/222533e69c154aa34e622122280b080ce50ce854.patch"
            }
        }
    },
```

The general idea is to replace references to `->record` with `->eventRecord`.

EDIT: I somehow smuggled laravel/pint into composer.json. It should pose no problem, but it was an accident.

**Comment by LuciusAB** [comment]
Also have this issue. Would be highly appreciated if it's possible to get it fixed, might be a while until V4 gets out?

**Comment by rickt96** [comment]
Recently faced the same issue.
My current workaround:
When the widget is referred in a resourcePage or CustomPage, i pass an additional parameter 'owner':
In the parent page
```
protected function getFooterWidgets(): array
    {
        return [
            MyCalendarWidget::make([
                "owned" => $this->record
            ])
        ];
    }
```

Then, inside MyCalendarWidget:
```
...
public Model $owned;
...
public function fetchEvents(array $fetchInfo): array
{
	return MyModel::query()
		// here $this->record is correctly defined, but i'm replacing it with my custom property
		->where('parent_id', $this->owned->id)
		->where('date_start', '>=', $fetchInfo['start'])
		->where('date_end', '<=', $fetchInfo['end'])
		->get()
		->map(function (ScheduleException $event) {
			...
		})
		->toArray();
}
```

Same usage inside the `getFormSchema()` or other functions
